util/types/datum_eval.go

```
@@ -55,11 +55,9 @@ func CoerceArithmetic(sc *variable.StatementContext, a Datum) (d Datum, err erro
 		}
 		d.SetMysqlDecimal(de)
 		return d, nil
-	case KindMysqlHex:
-		d.SetFloat64(a.GetMysqlHex().ToNumber())
-		return d, nil
-	case KindMysqlBit:
-		d.SetFloat64(a.GetMysqlBit().ToNumber())
+	case KindBinaryLiteral, KindMysqlBit:
+		val, _ := a.GetBinaryLiteral().ToInt()
```

error is not handled.